main/guides/js-programming/hardened-js.md

```diff
@@ -0,0 +1,388 @@
+# Hardened JavaScript
+
+::: tip Watch: Object-capability Programming in Secure Javascript (Aug 2019)
+
+_The first 15 minutes cover much of the material below.
+The last 10 minutes are Q&A._
+
+<iframe width="560" height="315" src="https://www.youtube.com/embed/YcWXqHPui_w" title="YouTube video player" frameborder="0" allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture" allowfullscreen></iframe>
+:::
+
+
+## Counter Example
+
+In case you thought JavaScript cannot be used to write
+reliable, secure smart contracts, we begin with this
+counter example. :)
+
+<<< @/snippets/test-hardened-js.js#makeCounter
+
+We'll unpack this a bit [below](#objects-and-the-maker-pattern), but for now,
+note the use of functions and records:
+
+ - `makeCounter` is a function
+ - Each call to `makeCounter` creates a new "instance":
+   - a new record with two properties, `incr` and `decr`, and
+   - a new `count` variable.
+ - The `incr` and `decr` properties are visible from
+   outside the object.
+ - The the `count` variable is encapsulated; only the
+   `incr` and `decr` methods can access it.
+ - Each of these instances is isolated from each other
+
+## Counter: Separation of Duties
+
+Suppose we want to keep track of the number of people
+inside a room by having an `entryGuard` count up when
+people enter the room and an `exitGuard` count down
+when people exit the room.
+
+<<< @/snippets/test-hardened-js.js#entryExit
```

Comment:
I agree they should be consistent.

But using `Far` rather than `harden` in the `makeCounter` example seems awfully awkward, given that the whole page is about **harden**ed JavaScript. I'm taking `E` out of the guard example.

I'm open to further discussion in a follow-on issue/PR.